File: 13-roman-to-integer/roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
       unordered_map<char, int> m = {{'I', 1}, {'V', 5}, {'X', 10}, {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}};
       int sum = 0;
       int n = s.size();
       int t = 0;
       while(t < n){
        if(t < n-1 && m[s[t]] < m[s[t+1]]){
            sum += m[s[t+1]] - m[s[t]];
            t += 2;
        }
        else {
            sum += m[s[t]];
            t ++;
        }
       }
       return sum;  
    }
};

File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> m = new HashMap();
        int i = 0;
        int sum = 0;
        int n = s.length();
        m.put('I', 1);
        m.put('V', 5);
        m.put('X', 10);
        m.put('L', 50);
        m.put('C', 100);
        m.put('D', 500);
        m.put('M', 1000);
        while(i < n){
        if(i < n - 1 && m.get(s.charAt(i)) < m.get(s.charAt(i+1))){
            sum += m.get(s.charAt(i+1)) - m.get(s.charAt(i));
            i +=2;
        }
        else{
            sum += m.get(s.charAt(i));
            i++;
        }
    }
    return sum;
    }
}